src/modules/trade/pure/TradeNumberInput/index.tsx

```diff
@@ -63,8 +63,7 @@ export function TradeNumberInput(props: TradeNumberInputProps) {
   // Initial setup of value
   useEffect(() => {
     validateInput(value ? value.toString() : '')
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-  }, [])
+  }, [value, validateInput])
```

Comment:
The code looks ok, but this change seems to break the input

I couldn't reproduce on video, but I saw at least twice in the parts input:

1. Focus on parts input
2. Use the arrow keys to move the value up
* Some times - don't know triggered by what - the value jumps back to a lower number of parts

Reply:
This is what I exactly was afraid of.
Fixed: https://github.com/cowprotocol/cowswap/pull/2952/commits/9c80334758416e5df335cf637b1214c6f0e8c01f
Thanks!